framework/console/controllers/MigrateController.php

```diff
@@ -312,11 +312,28 @@ protected function generateMigrationSourceCode($params)
             $table = mb_strtolower($matches[1], Yii::$app->charset);
         }
 
-        foreach ($foreignKeys as $column => $relatedTable) {
+        foreach ($foreignKeys as $column => $foreignKey) {
+            $relatedColumn = $foreignKey['column'];
+            $relatedTable = $foreignKey['table'];
+            if (empty($relatedColumn)) {
+                $relatedColumn = 'id';
+                $relatedTableSchema = $this->db->getTableSchema($relatedTable);
```

Comment:
https://github.com/yiisoft/yii2/pull/12770#discussion_r83872685

You can probably try-catch exception and fall back to default generation scenario...